### PR TITLE
Migrate all install scripts to python

### PR DESCRIPTION
**Pull request by cybre**
* Better install process
* One script for standard icon installs
* Custom scripts also migrated to python
* Fixed an issue with notifications
* Better error handling

### CODE REVIEW

**Comment by cybre** [comment]
This is a big one.
What do you guys think? @mank319 @quassy

File: scripts/custom/core.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
+import sys
+import os
+from os import symlink
```
Comment (quassy) [inline scripts/custom/core.py]: Not sure about the workings of Python but haven't you already imported os? Is this just for convenience so you can easily use `symlink.foobarmethod()`? Same with expanduser.

**Comment by quassy** [comment]
That's gonna take time to look at. 

I just have to check out `cybre:migrate-all-to-python` to test it right?

File: scripts/custom/core.py (hunk 1 above)
Comment (cybre) [inline scripts/custom/core.py]: It's just style, instinct :). It doesn't affect anything. I like having some parts of the library called like `module.fuction()` if I think I want to know at all times which module owns the function and some with just `function()` if I don't care what module the function comes from. Just like I said, I do it instinctively .

**Comment by cybre** [comment]
Yup!

**Comment by cybre** [comment]
@quassy Everything ok?

**Comment by quassy** [comment]
Running the installer seems to work fine. I didn't have any apps with indicators yet to actually test the scripts, but installing Skype or Spotify (the programs itself, not the elementary+ indicators) then killed Gala and now has somewhat borked my system (I have to kill Gala to use it) :D

**Comment by cybre** [comment]
Using the new Wingpanel?

**Comment by quassy** [comment]
I'm using elementary with the daily repo... "New Wingpanel"?

**Comment by cybre** [comment]
Wingpanel 0.4 (Complete rewrite with a new library and custom indicators)
screenshot from 2015-07-06 03 09 18
screenshot from 2015-07-06 03 09 37

I had some problems with Wingpanel (and gala) freezing when using qt apps with indicators.

**Comment by cybre** [comment]
It's in the staging PPA.

**Comment by quassy** [comment]
Pretty cool, my system still has 0.3 though. My issue seems to be related to VirtualBox / OpenGL graphic driver. And most likely that Skype relies on a too old Ubuntu version.

(Was about to ask about the PPA ;) )

**Comment by cybre** [comment]
So, safe to merge then?

**Comment by quassy** [comment]
I think so.

**Comment by cybre** [comment]
Great! Will merge in few minutes when I sort out the patches on the packaging side!

**Pull request merged**